task/net/src/server_vlan.rs

```diff
@@ -202,10 +208,23 @@ impl<'a> ServerImpl<'a> {
             }
             *ifaces_iter.next().unwrap() = Some(iface);
 
-            // Increment the MAC and IP addresses so that each VLAN has
-            // a unique address.
-            ipv6_addr.0[15] += 1;
-            mac.0[5] += 1;
+            // Increment the MAC and IP addresses based on the stride in the
+            // configuration block, so that each VLAN has a unique address.
+            //
+            // We take advantage of the fact that the top 3 octets are the OUI,
+            // and convert into a u32 to do the incrementing.
+            let next_mac = (u32::from_be_bytes(mac[2..].try_into().unwrap())
+                & 0xFFFFFF)
+                + mac_address_block.stride as u32;
+
+            // Per https://github.com/oxidecomputer/oana/#mac-addresses, we
+            // reserve `F0:00:00` and above for software stuff.
```

Comment:
This is only specific to the current OUI, this shouldn't be universal I think. If we get another block, it'd be totally fine to use that range.

Reply:
Interesting – do you think I should hard-code the OUI, e.g.
```rust
            const OXIDE_OUI: [u8; 3] = [0xa8, 0x40, 0x25];
            if mac[..3] == OXIDE_OUI && next_mac > 0xEFFFFF {
                panic!("MAC overflow: {:?}", mac);
            }
```
or should I check the U/L bit and assume that we want this masking if it's a universally administered address?

Reply:
Yeah, I think the hard coding approach makes sense. If we buy a second one or use something else for whatever reason then we won't want to apply that behavior. What you have looks good.